### PR TITLE
Add missing CLI options for dab configure, add, and update

### LINKED ISSUE #3410: Missing CLI options for dab configure, add and update (Group 1)
## Summary

Parent tracking issue for all missing CLI options in `dab configure`, `dab add`, and `dab update` commands. These are CLI coverage gaps where config properties exist in the schema but have no corresponding CLI option.

RCA:

New config properties (pagination, health, user-delegated-auth, etc.) were added to the schema and C# object model but the corresponding CLI options in ConfigureOptions.cs, EntityOptions.cs, and ConfigGenerator.cs were never created. A naming mismatch (cache.ttl vs cache.ttl-seconds, object-description vs description) also went unnoticed.

## Child Issues

- [ ] #3337 — Missing `runtime.pagination.max-page-size`
- [ ] #3338 — Missing `runtime.pagination.default-page-size`
- [ ] #3339 — Missing `runtime.pagination.next-link-relative`
- [ ] #3340 — Missing `runtime.host.max-response-size-mb`
- [ ] #3341 — Missing `runtime.health.*` (enabled, cache-ttl-seconds, max-query-parallelism, roles)
- [ ] #3342 — Missing `data-source-files`
- [ ] #3343 — Missing `runtime.telemetry.log-level`
- [ ] #3377 — `--cache.ttl` should be `--cache.ttl-seconds`
- [ ] #3383 — Missing `data-source.user-delegated-auth.provider`
- [ ] #3384 — Missing `data-source.health.threshold-ms`
- [ ] #3385 — Missing `data-source.health.enabled`
- [ ] #3386 — Dup of #3339 (`runtime.pagination.next-link-relative`)
- [ ] #3387 — Dup of #3340 (`runtime.host.max-response-size-mb`)
- [ ] #3388 — Missing entity `health.enabled`
- [ ] #3389 — Missing entity `cache.level`
- [ ] #3390 — Schema `source.object-description` should be `source.description`
- [ ] #3395 — `cosmosdb_postgresql` missing from `--database-type` help text

## Fix PR

- #3409

### PR DESCRIPTION
**Pull request by souvikghosh04**
## What

- Closes on #3410

Adds missing CLI options and fixes schema issues for `dab configure`, `dab add`, and `dab update` commands.

## Issues Fixed

| Issue | Description | Fix Location |
|-------|-------------|-------------|
| #3337 | Missing `runtime.pagination.max-page-size` | ConfigureOptions, ConfigGenerator |
| #3338 | Missing `runtime.pagination.default-page-size` | ConfigureOptions, ConfigGenerator |
| #3339 | Missing `runtime.pagination.next-link-relative` | ConfigureOptions, ConfigGenerator |
| #3340 | Missing `runtime.host.max-response-size-mb` | ConfigureOptions, ConfigGenerator |
| #3341 | Missing `runtime.health.*` (enabled, cache-ttl-seconds, max-query-parallelism, roles) | ConfigureOptions, ConfigGenerator |
| #3342 | Missing `data-source-files` | ConfigureOptions, ConfigGenerator |
| #3343 | Missing `runtime.telemetry.log-level` | ConfigureOptions, ConfigGenerator |
| #3377 | `--cache.ttl` should be `--cache.ttl-seconds` | EntityOptions |
| #3383 | Missing `data-source.user-delegated-auth.provider` | ConfigureOptions, ConfigGenerator |
| #3384 | Missing `data-source.health.threshold-ms` | ConfigureOptions, ConfigGenerator |
| #3385 | Missing `data-source.health.enabled` | ConfigureOptions, ConfigGenerator |
| #3386 | Dup of #3339 | (same fix) |
| #3387 | Dup of #3340 | (same fix) |
| #3388 | Missing entity `health.enabled` | EntityOptions, Utils, ConfigGenerator |
| #3389 | Missing entity `cache.level` | EntityOptions, Utils, ConfigGenerator |
| #3390 | Schema `object-description` to `description` | dab.draft.schema.json |
| #3395 | `cosmosdb_postgresql` missing from help text | ConfigureOptions |

- **ConfigureOptions.cs** - 15 new options
- **EntityOptions.cs** - Renamed `cache.ttl` to `cache.ttl-seconds`; added `cache.level`, `health.enabled`
- **AddOptions.cs / UpdateOptions.cs** - Pass new entity params
- **ConfigGenerator.cs** - Processing for all new options in configure, add, update flows
- **Utils.cs** - Updated `ConstructCacheOptions`; added `ConstructEntityHealthOptions`
- **dab.draft.schema.json** - Renamed `object-description` to `description`

## Tests

- [x] Unit tests
- New tests added

### CODE REVIEW

File: src/Cli/Utils.cs [hunk 1]
```diff
@@ -882,15 +892,44 @@ public static EntityGraphQLOptions ConstructGraphQLTypeDetails(string? graphQL,
...
+            {
+                cacheOptions = cacheOptions with { Enabled = isEnabled, TtlSeconds = ttl, UserProvidedTtlOptions = isCacheTtlUserProvided };
             }
 
-            if (cacheTtl is null)
+            if (level is not null)
             {
-                return cacheOptions with { Enabled = isEnabled };
+                cacheOptions = cacheOptions with { Level = level };
+            }
```
[the remark this answers is not in the document]
Comment (souvikghosh04) [reply src/Cli/Utils.cs]: Fixed

File: src/Cli/ConfigGenerator.cs [hunk 2]
```diff
@@ -988,6 +1001,33 @@ private static bool TryUpdateConfiguredRuntimeOptions(
...
+                }
+
+                if (options.RuntimePaginationNextLinkRelative != null)
+                {
+                    updatedPaginationOptions = updatedPaginationOptions with { NextLinkRelative = options.RuntimePaginationNextLinkRelative };
+                    _logger.LogInformation("Updated RuntimeConfig with runtime.pagination.next-link-relative as '{value}'", options.RuntimePaginationNextLinkRelative);
+                }
+
+                runtimeConfig = runtimeConfig! with { Runtime = runtimeConfig.Runtime! with { Pagination = updatedPaginationOptions } };
+            }
```
[the remark this answers is not in the document]
Comment (souvikghosh04) [reply src/Cli/ConfigGenerator.cs:1035]: rebuild PaginationOptions via constructor with merged values

File: src/Cli/ConfigGenerator.cs [hunk 3]
```diff
@@ -1062,6 +1124,48 @@ options.FileSinkRetainedFileCountLimit is not null ||
...
+
+                runtimeConfig = runtimeConfig! with
+                {
+                    Runtime = runtimeConfig.Runtime! with
+                    {
+                        Telemetry = runtimeConfig.Runtime!.Telemetry is not null
+                            ? runtimeConfig.Runtime!.Telemetry with { LoggerLevel = existingLevels }
+                            : new TelemetryOptions(LoggerLevel: existingLevels)
+                    }
+                };
```
[the remark this answers is not in the document]
Comment (souvikghosh04) [reply src/Cli/ConfigGenerator.cs:1182]: Fixed as suggested

File: src/Cli/ConfigGenerator.cs [hunk 4]
```diff
@@ -1062,6 +1124,48 @@ options.FileSinkRetainedFileCountLimit is not null ||
...
+                    string ns = parts[0].Trim();
+                    string levelStr = parts[1].Trim();
+                    if (!Enum.TryParse(levelStr, ignoreCase: true, out LogLevel level))
+                    {
+                        _logger.LogError("Invalid log level '{level}'. Allowed values: trace, debug, information, warning, error, critical, none.", levelStr);
+                        return false;
+                    }
+
+                    logLevels[ns] = level;
+                }
```
[the remark this answers is not in the document]
Comment (souvikghosh04) [reply src/Cli/ConfigGenerator.cs:1165]: added validation

File: src/Cli/Commands/EntityOptions.cs [hunk 5]
```diff
@@ -107,9 +111,15 @@ public EntityOptions(
         [Option("cache.enabled", Required = false, HelpText = "Specify if caching is enabled for Entity, default value is false.")]
         public string? CacheEnabled { get; }
 
-        [Option("cache.ttl", Required = false, HelpText = "Specify time to live in seconds for cache entries for Entity.")]
+        [Option("cache.ttl-seconds", Required = false, HelpText = "Specify time to live in seconds for cache entries for Entity.")]
```
Comment (RubenCerna2079) [inline src/Cli/Commands/EntityOptions.cs:114]: nit: I think we should also rename the variable to `CacheTtlSeconds`

File: src/Cli/Commands/ConfigureOptions.cs [hunk 6]
```diff
@@ -77,6 +89,7 @@ public ConfigureOptions(
             RollingInterval? fileSinkRollingInterval = null,
             int? fileSinkRetainedFileCountLimit = null,
             long? fileSinkFileSizeLimitBytes = null,
+            IEnumerable<string>? runtimeTelemetryLogLevel = null,
```
Comment (RubenCerna2079) [inline src/Cli/Commands/ConfigureOptions.cs:92]: I know they are not part of the bugs that were assigned to you, but we need to check if all the features from `runtime.telemetry` are here, since I don't see `application-insights`or `open-telemetry` in here.

File: src/Cli/Commands/ConfigureOptions.cs [hunk 7]
```diff
@@ -146,10 +173,12 @@ public ConfigureOptions(
             FileSinkRollingInterval = fileSinkRollingInterval;
             FileSinkRetainedFileCountLimit = fileSinkRetainedFileCountLimit;
             FileSinkFileSizeLimitBytes = fileSinkFileSizeLimitBytes;
+            // Telemetry Log Level
+            RuntimeTelemetryLogLevel = runtimeTelemetryLogLevel;
             ShowEffectivePermissions = showEffectivePermissions;
         }
 
-        [Option("data-source.database-type", Required = false, HelpText = "Database type. Allowed values: MSSQL, PostgreSQL, CosmosDB_NoSQL, MySQL.")]
+        [Option("data-source.database-type", Required = false, HelpText = "Database type. Allowed values: MSSQL, PostgreSQL, CosmosDB_NoSQL, CosmosDB_PostgreSQL, MySQL.")]
```
Comment (RubenCerna2079) [inline src/Cli/Commands/ConfigureOptions.cs]: I don't think we should add the `CosmosDB_PostgreSQL` data type, unless we are certain that we already support it.

File: src/Cli/ConfigGenerator.cs [hunk 8]
```diff
@@ -988,6 +1001,39 @@ private static bool TryUpdateConfiguredRuntimeOptions(
                 }
             }
 
+            // Pagination: MaxPageSize, DefaultPageSize, NextLinkRelative
+            if (options.RuntimePaginationMaxPageSize != null ||
+                options.RuntimePaginationDefaultPageSize != null ||
+                options.RuntimePaginationNextLinkRelative != null)
+            {
+                PaginationOptions existing = runtimeConfig?.Runtime?.Pagination ?? new();
```
Comment (RubenCerna2079) [inline src/Cli/ConfigGenerator.cs]: nit: I don't think we should call the variable `existing`

File: src/Cli/Commands/EntityOptions.cs (hunk 5 above)
Comment (souvikghosh04) [reply src/Cli/Commands/EntityOptions.cs:114]: Done — renamed `CacheTtl` to `CacheTtlSeconds` across EntityOptions, AddOptions, UpdateOptions, Utils.ConstructCacheOptions, ConfigGenerator, and all test call sites. Commit: 7c6c098.

File: src/Cli/Commands/ConfigureOptions.cs (hunk 6 above)
Comment (souvikghosh04) [reply src/Cli/Commands/ConfigureOptions.cs:92]: Good catch. The `application-insights` and `open-telemetry` options currently live in `AddTelemetryOptions.cs` (the `dab add-telemetry` command). Adding them to `dab configure` as well would be a new feature rather than a bug fix — I'll file a follow-up issue for that to keep this PR scoped to the bugs assigned.

File: src/Cli/Commands/ConfigureOptions.cs (hunk 7 above)
Comment (souvikghosh04) [reply src/Cli/Commands/ConfigureOptions.cs]: Agreed — removed `CosmosDB_PostgreSQL` from the HelpText. We can add it back once we've confirmed full support. Commit: 7c6c098.

File: src/Cli/ConfigGenerator.cs (hunk 8 above)
Comment (souvikghosh04) [reply src/Cli/ConfigGenerator.cs]: Done — renamed `existing` to `currentPagination` for clarity. Commit: 7c6c098.

File: src/Cli/Commands/ConfigureOptions.cs (hunk 6 above)
Comment (souvikghosh04) [reply src/Cli/Commands/ConfigureOptions.cs:92]: these live in dab add-telemetry. it is out of scope from this PR bug fix

File: src/Cli/Commands/ConfigureOptions.cs (hunk 7 above)
Comment (souvikghosh04) [reply src/Cli/Commands/ConfigureOptions.cs]: removed

File: src/Cli/Commands/ConfigureOptions.cs (hunk 6 above)
Comment (RubenCerna2079) [reply src/Cli/Commands/ConfigureOptions.cs:92]: We will have to discuss it with @JerryNixon

File: src/Cli.Tests/ConfigureOptionsTests.cs [hunk 9]
```diff
@@ -1590,5 +1590,269 @@ public void TestUpdateUserDelegatedAuthDatabaseAudience()
...
+        /// Tests adding runtime health options to a config that doesn't have a health section.
+        /// The enabled flag must be explicitly set for health options to persist in config.
+        /// Command: dab configure --runtime.health.enabled true --runtime.health.cache-ttl-seconds 10 --runtime.health.max-query-parallelism 2 --runtime.health.roles "admin,monitor"
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(true, 10, 2, new string[] { "admin", "monitor" }, DisplayName = "Set all runtime health options")]
+        [DataRow(false, null, null, null, DisplayName = "Disable runtime health")]
+        [DataRow(true, 30, null, null, DisplayName = "Enable health with custom cache-ttl")]
+        [DataRow(true, null, 8, null, DisplayName = "Enable health with custom max-query-parallelism")]
+        [DataRow(true, null, null, new string[] { "admin" }, DisplayName = "Enable health with roles")]
```
Comment (RubenCerna2079) [inline src/Cli.Tests/ConfigureOptionsTests.cs:1648]: Can we add a test case where `enabled` is null?

**Review by RubenCerna2079 — APPROVED** [review]
LGTM! Approving assuming comment will be addressed.

**Comment by anushakolan** [comment]
Can you also add the newly supported commands which you have added on top of each test, to the PR description?

**Comment by anushakolan** [comment]
Do we know why so many options were missing from the CLI?

File: src/Cli.Tests/ConfigureOptionsTests.cs [hunk 10]
```diff
@@ -1590,5 +1590,275 @@ public void TestUpdateUserDelegatedAuthDatabaseAudience()
             Assert.AreEqual(true, (bool?)userDelegatedAuthSection["enabled"]);
             Assert.AreEqual("EntraId", (string?)userDelegatedAuthSection["provider"]);
         }
```
Comment (anushakolan) [inline src/Cli.Tests/ConfigureOptionsTests.cs:1592]: Tests look great, can we add a small set of negative tests for documented ranges (page sizes < 1, threshold-ms <= 0, max-query-parallelism outside 1–8) + invalid telemetry level token (e.g. Default:NotALevel) to lock validation behavior?

**Comment by anushakolan** [comment]
Looks good to me. I’ve added a few comments that would be helpful to discuss or address. I don’t want to hold up this PR because of them, but please make sure to resolve and close those comments.

**Review by anushakolan — APPROVED** [review]

File: src/Cli.Tests/ConfigureOptionsTests.cs (hunk 10 above)
Comment (souvikghosh04) [reply src/Cli.Tests/ConfigureOptionsTests.cs:1592]: since this PR only wires up the CLI flags in existing constructors, this validation is not necessary within the scope of this PR.
